python/ray/tests/test_actor_failures.py

```diff
@@ -918,6 +918,10 @@ def f():
     class Actor:
         def exit(self):
             exit_actor()
+            raise AssertionError(
+                "This line should not be reached because non-async actor"
+                "should exit immediately after exit_actor is called."
+            )
```

Comment:
I don't think this assertion would actually catch a regression because this line might execute and then the `exit` method would still return a `RayActorError`. For any condition like this, you should verify that if you intentionally break the behavior (in this case, make it so that sync actors *don't* raise immediately), the test fails. I don't think it will in this case.

Instead, what you will need to do is add an "out of bounds" check -- so for example catch the `SystemExit` exception and then send a signal on an actor or write a tempfile like the other test cases. Then after the actor exits, assert that the exception was raised and therefore the out of band behavior happened.